src/Features/Core/Portable/UnusedReferences/UnusedReferencesRemover.cs

```diff
@@ -32,22 +34,24 @@ internal static class UnusedReferencesRemover
             CancellationToken cancellationToken)
         {
             var projects = solution.Projects
-                .Where(project => projectFilePath.Equals(project.FilePath, System.StringComparison.OrdinalIgnoreCase));
+                .Where(project => projectFilePath.Equals(project.FilePath, StringComparison.OrdinalIgnoreCase));
 
             HashSet<string> usedAssemblyFilePaths = new();
             HashSet<string> usedProjectFileNames = new();
 
-            foreach (var project in projects)
+            var getCompilationTasks = projects.Select(project => project.GetCompilationAsync(cancellationToken));
```

Comment:
so, ideally parallize on the project, not the compilation level.  Here's why, if you do it at the compilation level, then once we're done processing that compilation, we still can't let go of it due to the task holding onto it.  If we only hold onto the projec,t the project can release the compilation if not needed.

Reply:
I've decided to revert this change for now and will revisit performance in the future.